### DISCUSSION TITLE
MLR issue

### DISCUSSION THREAD

**Discussion by SaharRasoulpour** [firemodels/fds#13130] [Q&A]
Hello, I would appreciate your help with an issue I'm encountering. I have created a model of a wind tunnel that includes an infrared heater and some fuel particles. The infrared heater is modeled as a 9×9 cm surface maintained at a temperature of 971 K, producing a heat flux of 50 kW/m² at the surface.

The ignitor is modeled using a few non-reacting fuel particles held at a constant temperature of 1500°C, located 1.2 cm downstream of the fuel elements and centered 6 mm off the bottom. The fuel moisture content (FMC) of the vegetation is 70%.

My problem is that the total mass loss rate vs. time plot plateaus after 15 seconds. I need to determine whether this issue lies in my configuration or input file. Could you help me check if the problem is related to the ignitor?
[Needle.txt](https://github.com/user-attachments/files/16119706/Needle.txt)

**Comment by mcgratta** [comment]
```
&INIT  ID='pine needles',   PART_ID='pine needles', XB=0.255,0.345,0.1,0.19,-0.005,0.0, DENSITY = 550./
```
This is not how one should initiate particles. `DENSITY` in this case refers to the density of the gas, not the particles. In the current source code, `DENSITY` has been removed. Read the User's Guide chapter on Particles and Droplets, specifically on insertion.

**Comment by SaharRasoulpour** [comment]
Hello, I hope you are doing well. I have a couple of questions regarding my simulation. I set the initial temperature of my ignitor to 1500°C, but when I check the ignitor, it doesn't reach the specified temperature. Additionally, the infrared fire on top of the chamber has a temperature of 971K and a heat flux of 50 kW/m², exceeding the expected values. How can I control the temperature of my ignitor and the infrared fire? It is also important to note that the ignitor's temperature should reach 1500°C to ignite and transfer the heat to the needles for pyrolysis and combustion. Moreover, when I try to use Smoke view, I can't see the temperature slice. How should I check the temperature slice in Smoke view? Your assistance would be greatly appreciated.
[Wind Tunnel.txt](https://github.com/user-attachments/files/16286635/Wind.Tunnel.txt)

**Reply by drjfloyd** [reply]
Is this also you? https://github.com/firemodels/fds/discussions/12953#discussioncomment-10087625?

Same answer applies. TMP_GAS_FRONT has no meaning unless you have a heat conducting material for the SURF. Only the TMP_FRONT has meaning and RAMP_TMP_GAS_FRONT does not controll TMP_FRONT.

**Reply by SaharRasoulpour** [reply]
Thank you for your reply. No, I'm not #12953. Unfortunately, I am still facing an issue with the temperature slice in Smokeview. I can't see my slices. How should I proceed to resolve this problem? Additionally, if there are any specific settings or steps I need to follow to properly display the temperature slices, please let me know. Your assistance would be greatly appreciated.

**Reply by drjfloyd** [reply]
I can load your temperature slice without issue. Try installing the latest test bundle of smokeview:

https://github.com/firemodels/test_bundles/releases

You are mixing combustion models:

&REAC ID='WOOD', FUEL='METHANE', SOOT_YIELD=0.02, C=1.0, H=4.0, HEAT_OF_COMBUSTION=55500.
      SPEC_ID_NU= 'METHANE', 'AIR' , 'PRODUCT'
      NU= -1, -1.98, 10.44857143/

Either use C,H,O,SOOT_YIELD, or NU but not both.  If you use C,H,O you don't need to define AIR and PRODUCT.  SOOT_YIELD has no meaning with NU.  You may wish to review the many examples of REAC inputs in the User's Guide.

**Reply by SaharRasoulpour** [reply]
Thank you for your reply. When I try to define HT3D for my pine needle, I encounter a segmentation fault error. How should I address this type of error?


!NEEDLE:
&SURF ID                        = 'PINE NEEDLE'
      MATL_ID                   = 'PINE'
      MOISTURE_FRACTION         = 0.7
      DRAG_COEFFICIENT          = 0.375 
      TMP_FRONT                 = 1500
      HT3D                      = T
      COLOR                     = 'GREEN'
      SURFACE_VOLUME_RATIO      = 8366.
      BURN_AWAY                 = .TRUE.
      THICKNESS                 = 0.005
      WIDTH                     = 0.09
      LENGTH                    = 0.09/

&MATL ID                        = 'PINE'
      DENSITY                   = 550.
      CONDUCTIVITY              = 1.69E-1
      SPECIFIC_HEAT             = 1.8
      N_REACTIONS               = 1
      A                         = 3.63E4
      E                         = 60279.83355
      MATL_ID                   = 'CHAR'        
      NU_MATL                   = 0.25
      SPEC_ID                   = 'METHANE'
      NU_SPEC                   = 0.75
      HEAT_OF_REACTION          = 1000/

&MATL ID                        = 'MOISTURE'
      DENSITY                   = 1000.
      CONDUCTIVITY              = 0.1
      SPECIFIC_HEAT             = 4.184
      N_REACTIONS               = 1
      A                         = 5.13E10
      E                         = 88000.24252
      NU_SPEC                   = 1.0
      SPEC_ID                   = 'WATER VAPOR'
      HEAT_OF_REACTION          = 2500. /

&MATL ID                        = 'CHAR'
      DENSITY                   = 300.
      CONDUCTIVITY              = 50.052
      SPECIFIC_HEAT             = 1.0/

&OBST XB = 0.255,0.345,0.085,0.175,0.0035,0.0085, COLOR = 'GREEN', SURF_ID = 'PINE NEEDLE', BULK_DENSITY=42.9/ !PINE BED

**Reply by mcgratta** [reply]
I would need to see the entire file, but I also question why you want to use 3D heat conduction in an object that can be considered thermally thin.

**Reply by SaharRasoulpour** [reply]
I need 3D heat conduction on my pine needle surface to observe how the ignitor's temperature spreads and affects the temperature distribution on the surface.


&HEAD CHID='NEEDLE_FMC70', TITLE='Wind Tunnel Setup' /
&TIME T_END=10., DT=0.01, WALL_INCREMENT=1 /
&MISC TMPA=20./
&RADI RADIATION=.TRUE./
&MESH IJK=60,26,11, XB=0.0,0.6, 0.0,0.26,0.0,0.110 /

&SPEC ID= 'METHANE'/
&SPEC ID='WATER VAPOR' /
&SPEC ID='CARBON DIOXIDE', LUMPED_COMPONENT_ONLY = .TRUE. /
&SPEC ID='SOOT', LUMPED_COMPONENT_ONLY = .TRUE. /
&SPEC ID='NITROGEN', LUMPED_COMPONENT_ONLY = .TRUE./
&SPEC ID='OXYGEN', LUMPED_COMPONENT_ONLY = .TRUE./
&SPEC ID = 'AIR', SPEC_ID = 'OXYGEN','NITROGEN', VOLUME_FRACTION = 1,3.761904762, BACKGROUND=.TRUE./
&SPEC ID = 'PRODUCT', SPEC_ID = 'CARBON DIOXIDE','WATER VAPOR','SOOT','NITROGEN',  VOLUME_FRACTION = 0.09379272627, 0.1914137271, 0.001914137271, 0.7128794094/

&REAC ID='WOOD', FUEL='METHANE', SOOT_YIELD=0.02, C=1.0, H=4.0, HEAT_OF_COMBUSTION=55500. 
      SPEC_ID_NU= 'METHANE', 'AIR' , 'PRODUCT'
      NU= -1, -1.98, 10.44857143/

&SURF ID='WALL', TMP_FRONT=20.0/

&VENT MB='XMIN',SURF_ID='OPEN' /
&OBST XB=0.0,0.6,0.0,0.0,0.0,0.110, COLOR='WHITE', TRANSPARENCY=0.5, SURF_ID='WALL' /!FRONT
&OBST XB=0.0,0.6,0.26,0.26,0.0,0.110, COLOR='WHITE', TRANSPARENCY=0.5, SURF_ID='WALL'  /!BACK
&OBST XB=0.0,0.6,0.0,0.085,0.110,0.110, COLOR='WHITE', TRANSPARENCY=0.5 , SURF_ID='WALL' /!UP_front
&OBST XB=0.0,0.6,0.175,0.26,0.110,0.110, COLOR='WHITE', TRANSPARENCY=0.5 , SURF_ID='WALL' /!UP_back
&OBST XB=0.0,0.255,0.085,0.175,0.110,0.110, COLOR='WHITE', TRANSPARENCY=0.5 , SURF_ID='WALL' /!UP_left
&OBST XB=0.345,0.6,0.085,0.175,0.110,0.110, COLOR='WHITE', TRANSPARENCY=0.5 , SURF_ID='WALL' /!UP_right
&OBST XB=0.0,0.6,0.0,0.26,0.0,0.0, COLOR='WHITE', TRANSPARENCY=0.1 , SURF_ID='WALL' /!bottom
!&OBST XB=0.0,0.6,0.0,0.085,0.0,0.0, COLOR='WHITE', TRANSPARENCY=0.5 , SURF_ID='WALL' /!bottom-front
!&OBST XB=0.0,0.6,0.175,0.26,0.0,0.0, COLOR='WHITE', TRANSPARENCY=0.5 , SURF_ID='WALL' /!bottom-back
!&OBST XB=0.0,0.255,0.085,0.175,0.0,0.0, COLOR='WHITE', TRANSPARENCY=0.5 , SURF_ID='WALL' /!bottom-left
!&OBST XB=0.345,0.6,0.085,0.175,0.0,0.0, COLOR='WHITE', TRANSPARENCY=0.5 , SURF_ID='WALL' /!bottom-right


&SURF ID='AIR INLET', VEL=-1.0, SPEC_ID='AIR'/
&VENT XB=0.6,0.6,0.0,0.26,0.0,0.110, SURF_ID='AIR INLET', COLOR='RED', TRANSPARENCY=0.5 /

!HEATER
&SURF ID='FIRE', TMP_FRONT=698.0 , RGB= 255,102,102/ 
&VENT XB=0.255,0.345,0.085,0.175,0.110,0.110, SURF_ID='FIRE' /

!IGNITOR:
&SURF ID                        ='IGNITOR ELEMENTS'
        SURFACE_VOLUME_RATIO    = 8366.0
        TMP_FRONT               = 1500.
        DRAG_COEFFICIENT        = 0.1
        RAMP_T                  ='RAMP IGN'
        THICKNESS               = 0.005
        WIDTH                   = 0.09
        LENGTH                  = 0.0075/
 
&RAMP ID='RAMP IGN', T=0, F=0 /
&RAMP ID='RAMP IGN', T=0.5, F=1 /
&RAMP ID='RAMP IGN', T=250, F=1 /
&RAMP ID='RAMP IGN', T=500, F=0.5 /
&OBST XB = 0.243,0.2505,0.085,0.175,0.0035,0.0085, COLOR = 'RED', SURF_ID = 'IGNITOR ELEMENTS'/ !IGNITOR

!NEEDLE:
&SURF ID                        = 'PINE NEEDLE'
      MATL_ID                   = 'PINE'
      MOISTURE_FRACTION         = 0.7
      DRAG_COEFFICIENT          = 0.375 
      TMP_FRONT                 = 1500
      EXTERNAL_FLUX             = 50
      COLOR                     = 'GREEN'
      SURFACE_VOLUME_RATIO      = 8366.
      BURN_AWAY                 = .TRUE.
      THICKNESS                 = 0.005
      WIDTH                     = 0.09
      LENGTH                    = 0.09/

&MATL ID                        = 'PINE'
      DENSITY                   = 550.
      CONDUCTIVITY              = 1.69E-1
      SPECIFIC_HEAT             = 1.8
      N_REACTIONS               = 1
      A                         = 3.63E4
      E                         = 60279.83355
      MATL_ID                   = 'CHAR'        
      NU_MATL                   = 0.25
      SPEC_ID                   = 'METHANE'
      NU_SPEC                   = 0.75
      HEAT_OF_REACTION          = 1000/

&MATL ID                        = 'MOISTURE'
      DENSITY                   = 1000.
      CONDUCTIVITY              = 0.1
      SPECIFIC_HEAT             = 4.184
      N_REACTIONS               = 1
      A                         = 5.13E10
      E                         = 88000.24252
      NU_SPEC                   = 1.0
      SPEC_ID                   = 'WATER VAPOR'
      HEAT_OF_REACTION          = 2500. /

&MATL ID                        = 'CHAR'
      DENSITY                   = 300.
      CONDUCTIVITY              = 50.052
      SPECIFIC_HEAT             = 1.0/

&OBST XB = 0.255,0.345,0.085,0.175,0.0035,0.0085, COLOR = 'GREEN', SURF_ID = 'PINE NEEDLE', BULK_DENSITY=42.9/ !PINE BED
&PART ID='PINE NEEDLE', SAMPLING_FACTOR=1, SURF_ID='PINE NEEDLE',STATIC=T/

!DUMPS:
&DUMP DT_SLCF = 2, DT_PART = 1 /
&DUMP PLOT3D_QUANTITY(1:2) = 'TEMPERATURE', 'VELOCITY', DT_PL3D = 2.0, WRITE_XYZ = .TRUE./
!SLICES:
&SLCF PBY=0.13, QUANTITY='TEMPERATURE', VECTOR=.TRUE. /
&SLCF PBZ=0.0025, QUANTITY='TEMPERATURE', VECTOR=.TRUE. /

!DEVCES:
!TEMPERATURE:
&DEVC XB=0.0,0.6, 0.0,0.26,0.0,0.110, QUANTITY='TEMPERATURE', ID='TEMPERATURE' /
&DEVC XYZ=0.247,0.13,0.006, QUANTITY='WALL TEMPERATURE', ID='IGNITOR TMP', IOR=3 /
&DEVC XYZ=0.3,0.13,0.110, QUANTITY='WALL TEMPERATURE', ID='HEATER TMP', IOR=-3 /
&DEVC XYZ=0.3,0.13,0.0025, QUANTITY='WALL TEMPERATURE', ID='PINE BED TMP', IOR=3 /

&DEVC XYZ=0.255,0.13,0.006, QUANTITY='WALL TEMPERATURE', ID='PINE BED TMP2', IOR=3 /
&DEVC XYZ=0.255,0.13,0.006, QUANTITY='TEMPERATURE', ID='TEMPERATURE PINE', SURF_ID= 'PINE NEEDLE' /


&TAIL /

**Reply by mcgratta** [reply]
The case is running for me with the latest source code. When does you case fail and what version of FDS are you using?

**Reply by SaharRasoulpour** [reply]
This case is running correctly, but when I try to use HT3D in SURF_ID='PINE NEEDLE', I encounter an error.

**Reply by mcgratta** [reply]
```
ERROR(369): SURF PINE NEEDLE  cannot have both HT3D and BURN_AWAY. (CHID: NEEDLE_FMC70)
```

**Reply by SaharRasoulpour** [reply]
Even after removing BURN_AWAY, the error persists.

**Reply by mcgratta** [reply]
The case is running for me. Are you using the latest nightly build?

**Reply by SaharRasoulpour** [reply]
Yes, I am using the latest version of FDS.

**Reply by drjfloyd** [reply]
To be clear, by latest build we mean the version here: https://github.com/firemodels/test_bundles/releases

**Reply by SaharRasoulpour** [reply]
My main concern with this simulation is that when the ignitor and heater spread heat to the pine needle bed, the pine needles don't seem to reach desired temperature. Because of this, I used HT3D to observe conduction in the pine needle bed and monitor the mass loss rate. I defined OBST for the pine needle bed, but even after running the simulation for 250 seconds, the temperature in the pine needle bed still fluctuates around 300°C. I saw similar cases where OBST was used for vegetation or pine beds. Is there another method I could use? I think my pine needle bed is designed to feel the surrounding temperature but doesn't conduct this heat internally. What do you suggest I do to observe a temperature difference within the pine needle bed?

**Reply by mcgratta** [reply]
HT3D is not going to help you. You cannot simulate both a fire spreading in a pine needle bed and detailed 3D thermal conduction within the needles themselves. I suggest you run and study one of the cases called `USFS Corsica`. They are described in the FDS Validation Guide. The input files are [here](https://github.com/firemodels/fds/tree/master/Validation/USFS_Corsica/FDS_Input_Files).

**Reply by SaharRasoulpour** [reply]
Thank you for your reply; it was really helpful. However, I've encountered an issue where the temperature of the pine needles reaches 800°C between 20 to 40 seconds, then suddenly decreases before increasing again. This pattern occurs at different times as well. When I check the ignitor, its temperature remains steady at 1500°C, indicating no reduction. Which parameter should I control to prevent this temperature fluctuation in the pine needles?

**Reply by mcgratta** [reply]
I suggest that you run a simple single mesh case with one pine needle. Use this case to check your model, model functionality, and if you see a potential bug, we can use this case for debugging.

**Reply by SaharRasoulpour** [reply]
If you look at the plot generated from the FDS simulation, you'll see that the temperature on the left-hand side (near the ignitor) rises and then suddenly drops to 100 degrees Celsius. I am uncertain about the cause of the temperature decrease in the bed. Could it be due to the thermal properties of the pine? I have run this case multiple times, including with a simpler setup, but the issue persists. I have uploaded the plot along with the FDS input files. Your insight and suggestions would be greatly appreciated to help me correct this and achieve accurate results.
![image](https://github.com/user-attachments/assets/ca7dcb55-9fdc-4ad4-95ff-33536a51d6fa)
[sample.txt](htt
[vegetation_model.txt](https://github.com/user-attachments/files/16433592/vegetation_model.txt)

[sample.txt](https://github.com/user-attachments/files/16433607/sample.txt)

**Reply by drjfloyd** [reply]
You are putting one pine needle per cell with your inputs.

**Reply by SaharRasoulpour** [reply]
Thank you for your response and consideration. I encountered an issue where, after setting N_particle_per_cell = 8, the fire propagation on the bed does not drop below 100°C after 25 seconds fortunately. The fire progresses well after that, but I’m concerned that between 50 seconds, and continuing until around 160 seconds, there’s unexpected ongoing combustion, as shown in the plot. I’ve tried adjusting the kinetic properties, but the problem persists during this interval. Do you suggest that I focus on the kinetic properties, or could this issue be related to my configuration or combustion settings?
![image](https://github.com/user-attachments/assets/936b4420-82d2-4f12-8746-143ac9c2098d)

[Wind_Tunnel.txt](https://github.com/user-attachments/files/16663397/Wind_Tunnel.txt)
[vegetation_model.txt](https://github.com/user-attachments/files/16663398/vegetation_model.txt)

**Reply by SaharRasoulpour** [reply]
When I defined the OBST for my pine bed, that region improved, but the peak temperature increased significantly. If I want to control the peak temperature—reducing it from 800°C to 600°C—which parameter should I focus on? Is it correct to define OBST for my pine bed? I added OBST because some FDS samples included OBST in the pine bed. My plots are close to what I want, but I still have some issues. In my last message, my sample didn’t have OBST for the pine bed, but in this case, it does.
!PINE BED:
&SURF ID                   = 'Pine Needles'
      MATL_ID              = 'GENERIC VEGETATION'
      MOISTURE_FRACTION    = 0.7
      SURFACE_VOLUME_RATIO = 8366.
      LENGTH               = 0.09
      GEOMETRY             = 'CYLINDRICAL' /

&PART ID             = 'Needles' 
      DRAG_COEFFICIENT =0.375
      COLOR          = 'GREEN'
      SURF_ID        = 'Pine Needles' 
      QUANTITIES     = 'PARTICLE TEMPERATURE','PARTICLE MASS'
      STATIC         = .TRUE.
      EMBER_PARTICLE       = T
      EMBER_DENSITY_THRESHOLD = 70. 
      EMBER_VELOCITY_THRESHOLD = 0.
      TRACK_EMBERS         = F /

&INIT PART_ID              = 'Needles'
      XB                   = 0.255,0.345,0.085,0.175,0,0.005
      N_PARTICLES_PER_CELL = 8
      CELL_CENTERED        = .TRUE.
      DRY                  = T /
&OBST XB=0.255,0.345,0.085,0.175,0,0.005, COLOR='WHITE', TRANSPARENCY=0.5 , SURF_ID='Pine Needles', BULK_DENSITY=42.9 /

![image](https://github.com/user-attachments/assets/32813ca0-2b2e-4908-8565-e8ac41501b93)
